core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -959,7 +959,7 @@ public String getSearchName() {
     @Restricted(NoExternalUse.class)
     public Object getTarget() {
         if (!SKIP_PERMISSION_CHECK) {
-            if (!getACL().hasPermission(Item.DISCOVER)) {
+            if (!hasPermission(Item.DISCOVER)) {
```

Comment:
Just cleaning up while I am here, especially in case we decide to remove https://github.com/jenkinsci/jenkins/blob/9116ce84feb81d6c709ec425dde131baacf543c8/core/src/main/java/hudson/security/ACL.java#L134-L136

Reply:
the code 3 lines down still does `getACL().checkPermission(Item.READ)`  slightly confused here?

Reply:
* `READ` ⇒ `this`
* `!DISCOVER` ⇒ `AccessDeniedException`
* `DISCOVER && !READ` ⇒ `null`

Reply:
I was not clear.

with this change

```java
    @Override
    @Restricted(NoExternalUse.class)
    public Object getTarget() {
        if (!SKIP_PERMISSION_CHECK) {
            if (!hasPermission(Item.DISCOVER)) {
                return null;
            }
            getACL().checkPermission(Item.READ);
        }
        return this;
    }
```

why is this not

```
    @Override
    @Restricted(NoExternalUse.class)
    public Object getTarget() {
        if (!SKIP_PERMISSION_CHECK) {
            if (!hasPermission(Item.DISCOVER)) {
                return null;
            }
            checkPermission(Item.READ);
        }
        return this;
    }
```

I mean the line with the permission check for `Item.READ` was still calling `getACL` which seems strange?

Reply:
Ah OK, yes that looks like an oversight.